controllers/managedclusterForCGU_controller.go

```diff
@@ -206,8 +206,9 @@ func (r *ManagedClusterForCguReconciler) newClusterGroupUpgrade(
 		Name:      cluster.Name,
 		Namespace: ztpInstallNS,
 	}
+	enable := true // default
 	cguSpec := ranv1alpha1.ClusterGroupUpgradeSpec{
-		Enable:          true, // default
+		Enable:          &enable,
```

Comment:
There is an issue in controller-runtime: https://github.com/clastix/capsule/issues/342, https://github.com/kubernetes-sigs/kubebuilder/issues/2109
If we are trying to update the CR, the boolean non pointer values are overwritten with the default value. Angie had the same issues with deleteObjectsOnCompletion and changed it to a pointer because of that. I changed it to avoid the same issue in the future and to also be consistent.